### DISCUSSION TITLE
Ubuntu - installation test failed. What next?

### DISCUSSION THREAD

**Discussion by stephan0h** [AviSynth/AviSynthPlus#213] [Q&A]
Hi, 
Installed according to https://github.com/AviSynth/AviSynthPlus/blob/master/distrib/docs/english/source/avisynthdoc/contributing/posix.rst - and now it seems I'm stuck here:

ffplay -i test.avi
--> test.avi: Invalid data found when processing input

Content of test.avi:
Version()

I'm running Ubuntu 18.04.5 LTS.

Some hints how to check my installation would be very appreciated :-)

Thanks,
Stephan

**Comment by qyot27** [comment]
The extension for AviSynth scripts is `.avs`.  Using `.avi` is telling it to use the wrong demuxer (although you can override ff[mpeg|play|probe]'s default demuxer probing by forcing the format with *-f avisynth*).

**Reply by stephan0h** [reply]
Oh, thanks, silly me ;-)

But now I tried this:
  ffplay -i test.avs -f avisynth

And get this:
  libavutil      55. 78.100 / 55. 78.100
  libavcodec     57.107.100 / 57.107.100
  libavformat    57. 83.100 / 57. 83.100
  libavdevice    57. 10.100 / 57. 10.100
  libavfilter     6.107.100 /  6.107.100
  libavresample   3.  7.  0 /  3.  7.  0
  libswscale      4.  8.100 /  4.  8.100
  libswresample   2.  9.100 /  2.  9.100
  libpostproc    54.  7.100 / 54.  7.100
  test.avs: Unknown error occurred
    nan    :  0.000 fd=   0 aq=    0KB vq=    0KB sq=    0B f=0/0   

How could I find out about this unknown error?

Thanks ! :-)

**Comment by qyot27** [comment]
That error usually occurs when the `libavisynth.so` symlink can't be found; there are some circumstances where it doesn't get installed even when the main library and SOVERSION symlink do, but I haven't found a reproduceable set of conditions to show it happening consistently.  Check /usr/local/lib and see if all three files exist (`libavisynth.so.3.7.0`, `libavisynth.so.8`, and `libavisynth.so`, currently).

**Reply by stephan0h** [reply]
Well this seems to be ok. It looks like this:

stephan@rechenmonster:/usr/local/lib$ ls -als
insgesamt 12208
    4 drwxr-xr-x  8 root root      4096 Mär 25 20:38 .
    4 drwxr-xr-x 11 root root      4096 Jän  1 08:24 ..
    4 drwxr-xr-x  2 root root      4096 Mär 25 20:37 avisynth
    0 lrwxrwxrwx  1 root root        16 Mär 25 20:38 libavisynth.so -> libavisynth.so.8
12176 -rw-r--r--  1 root root  12466448 Mär 25 19:21 libavisynth.so.3.7.0
    0 lrwxrwxrwx  1 root root        20 Mär 25 20:37 libavisynth.so.8 -> libavisynth.so.3.7.0

And that's the version of ffplay:

stephan@rechenmonster:/usr/local/lib$ ffplay -v
ffplay version 3.4.8-0ubuntu0.2 Copyright (c) 2003-2020 the FFmpeg developers
  built with gcc 7 (Ubuntu 7.5.0-3ubuntu1~18.04)

I guess maybe it's the version of ffmpeg and ffplay. Just checked the timestamp of the binary files and it is July 16th. But I built it yesterday.

btw: if it's ok I could extend the installation docs with my findings - once I get there.

**Reply by stephan0h** [reply]
Ok, now I tried with the version of ffplay in ~/ffmpeg_build/bin - and it worked. So apparently the binaries where not installed globally. 
Thanks for you support!
